sdk/core/azure-core/azure/core/paging/_paging_method.py

```diff
@@ -0,0 +1,241 @@
+            return getattr(deserialized, item_name)
+        raise ValueError(
+            "The response object does not have property '{}' to extract element list from".format(item_name)
+        )
+
+    def mutate_list(self, pipeline_response, list_of_elem, cls=None):
+        # type: (HttpResponse, Iterable[ReturnType], Optional[Callable]) -> Iterable[ReturnType]
+        """Mutate list of elements in current page, i.e. if users passed in a cls calback
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param list[object] list_of_elem: The iterable of items we will return to users
+         in a page.
+        :return: The final list of iterable items we will return to users
+        :rtype: iterable[object]
+        """
+        if cls:
+            list_of_elem = cls(list_of_elem)
+        return iter(list_of_elem)
+
+    def get_continuation_token(self, pipeline_response, deserialized, continuation_token_location=None):
+        # type: (HttpResponse, ResponseType, Optional[str]) -> Any
+        """Get the continuation token from the current page. This operation returning None signals the end of paging.
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param object deserialized: The deserialized pipeline_response
+        :return: The continuation token. Can be any value, and will be passed as-is to :func:`get_next_request`.
+        :rtype: any
+        """
+        if continuation_token_location:
+            if not hasattr(deserialized, continuation_token_location):
+                raise ValueError(
+                    "The response object does not have property '{}' to extract continuation token from".format(
+                        continuation_token_location
+                    )
+                )
+            return getattr(deserialized, continuation_token_location)
+        return None
+
+class ContinueWithNextLink(ContinueWithCallback):  # pylint: disable=too-many-instance-attributes
+
+    def __init__(self, path_format_arguments=None, **kwargs):
+        """Most common paging method. Uses the continuation token as the URL for the next call.
+        """
+        def _next_request_callback(continuation_token, initial_request, client):
+            request = initial_request
+            next_link = continuation_token
+            if self._path_format_arguments:
+                next_link = client.format_url(next_link, **self._path_format_arguments)
+            else:
+                next_link = client.format_url(next_link)
+            request.url = next_link
+            return request
+
+        super(ContinueWithNextLink, self).__init__(
+            next_request_callback=_next_request_callback, **kwargs
+        )
+        self._path_format_arguments = path_format_arguments
```

Comment:
Do we really need this instance attribute? The argument passed in is already captured in the `_next_request_callback`, so is there a specific need to access if from `self`? I don't see it being mutated anywhere along the way...

Reply:
good point, won't catch it